python/cudf/cudf/_lib/unaryops.pyx

```diff
@@ -157,9 +159,13 @@ def nans_to_nulls(Column py_col):
         result = cpp_unaryops.nans_to_nulls(c_col[0])
 
     mask = None
-    if result.second:
+    if result.second > 0:
         mask_ptr = int(<uintptr_t>result.first)
```

Comment:
This is still leaked because nothing frees it. Need to call rmmFree on it.

Reply:
Right you are.